src/tools.rs

```diff
@@ -482,6 +484,35 @@ pub async fn read_dir(path: &Path) -> Result<Vec<fs::DirEntry>> {
     Ok(res)
 }
 
+#[cfg(test)]
+static SYSTEM_TIME_SHIFT: RwLock<Duration> = RwLock::new(Duration::new(0, 0));
+
+/// Fake struct for mocking `SystemTime::now()` etc. for test purposes. You still need to use
+/// `std::time::SystemTime` as a struct representing a system time.
+pub(crate) struct SystemTime();
+
+impl SystemTime {
+    pub const UNIX_EPOCH: std::time::SystemTime = std::time::SystemTime::UNIX_EPOCH;
+
+    #[allow(unused)]
```

Comment:
I think that if we move it to a separate crate, we need to truly mock the whole `SystemTime`, not only `now()`, making such a "crutch" too public doesn't look good. As for `deltachat-ratelimit`, in fact it needs `Instant`. So, if we want to test `deltachat-ratelimit` with time jumps, we can actually use tokio's `Instant` in the test cfg, it has `advance()`, `pause()` etc.

Reply:
> making such a "crutch" too public doesn't look good

I don't think that would be a problem - many Rust crates (even libraries with stability guarantees and lots of users) have internal "helper" crates that are not meant to be used by others, and without a nice API.

> As for `deltachat-ratelimit`, in fact it needs `Instant`.

That's indeed a reason to keep things simple for now and only mock within the `deltachat` crate itself. Still, I think that it would be good to mention in a comment somewhere that, unfortunately, this is only mocking in the `deltachat` crate itself and in order to also mock in `deltachat-ratelimit` and others, one would need to make an extra crate `deltachat-time` which would then also include `Instant`.

Reply:
Moved it into a separate crate and also removed `elapsed()` -- it's trivial and can be added later if will be really needed.